cmake/kokkos_arch.cmake

```diff
@@ -569,26 +577,18 @@ ENDFUNCTION()
 
 #These will define KOKKOS_AMDGPU_ARCH_FLAG
 #to the corresponding flag name if ON
-CHECK_AMDGPU_ARCH(VEGA900 gfx900) # Radeon Instinct MI25
-CHECK_AMDGPU_ARCH(VEGA906 gfx906) # Radeon Instinct MI50 and MI60
-CHECK_AMDGPU_ARCH(VEGA908 gfx908) # Radeon Instinct MI100
-CHECK_AMDGPU_ARCH(VEGA90A gfx90a) # Radeon Instinct MI200
+FOREACH(ARCH IN LISTS SUPPORTED_AMD_ARCHS)
+  LIST(FIND SUPPORTED_AMD_ARCHS ${ARCH} LIST_INDEX)
+  LIST(GET CORRESPONDING_AMD_FLAGS ${LIST_INDEX} FLAG)
+  CHECK_AMDGPU_ARCH(${ARCH} ${FLAG})
+ENDFOREACH()
 
-IF(KOKKOS_ENABLE_HIP AND NOT AMDGPU_ARCH_ALREADY_SPECIFIED)
-  IF(KOKKOS_CXX_COMPILER_ID STREQUAL HIPCC)
-    FIND_PROGRAM(ROCM_ENUMERATOR rocm_agent_enumerator)
-    EXECUTE_PROCESS(COMMAND ${ROCM_ENUMERATOR} OUTPUT_VARIABLE GPU_ARCHS)
-    STRING(LENGTH "${GPU_ARCHS}" len_str)
-    # enumerator always output gfx000 as the first line
-    IF(${len_str} LESS 8)
-      MESSAGE(SEND_ERROR "HIP enabled but no AMD GPU architecture currently enabled. "
-                         "Please enable one AMD GPU architecture via -DKokkos_ARCH_{..}=ON'.")
-    ENDIF()
-  ELSE()
-    MESSAGE(SEND_ERROR "HIP enabled but no AMD GPU architecture currently enabled. "
-                       "Please enable one AMD GPU architecture via -DKokkos_ARCH_{..}=ON'.")
-  ENDIF()
-ENDIF()
+MACRO(SET_AND_CHECK_AMD_ARCH ARCH FLAG)
+  KOKKOS_SET_OPTION(ARCH_${ARCH} ON)
+  CHECK_AMDGPU_ARCH(${ARCH} ${FLAG})
+  LIST(APPEND KOKKOS_ENABLED_ARCH_LIST ${ARCH})
+  SET(KOKKOS_ENABLED_ARCH_LIST  ${KOKKOS_ENABLED_ARCH_LIST})
```

Comment:
??

Reply:
good catch ... forgot to delete when converting the function into a macro ... deleted the set in [50c17e](https://github.com/kokkos/kokkos/pull/5327/commits/50c17e17b94b5114eb9bf5eeb728ba25c4c885f5)